Generic test `arguments` deprecations and behavior change flags (#7676)
fishtown-analytics/docs.getdbt.com: website/docs/reference/deprecations.md

The following are deprecation warnings in dbt today and the associated version number in which they first appear.

### ArgumentsPropertyInGenericTestDeprecation

dbt has deprecated the ability to specify a custom top-level property called `arguments` on generic tests. This deprecation warning is only raised when the behavior flag `require_generic_test_arguments_property` is set to `False`.

#### ArgumentsPropertyInGenericTestDeprecation warning resolution

If you previously had a property called `arguments` on custom generic tests:

<File name='model.yml'>

```yaml
models:
  - name: my_model_with_generic_test
    data_tests:
      - my_custom_generic_test:
          arguments: [1,2,3]

>           expression: "order_items_subtotal = subtotal"

Reviewer: ```suggestion
          expression: "order_items_subtotal = subtotal"
```

```

> 

Reviewer: Oops this needs to be closed
```suggestion
</File>
```

You should flip the `require_generic_test_arguments_property` flag to `True` and nest any keyword arguments to your test under the new `arguments` property:

<File name='model.yml'>

```yaml
models:
  - name: my_model_with_generic_test
    data_tests:
      - my_custom_generic_test:
          arguments: 

>             arguments: [1,2,3]

Reviewer: @luna-bianca Should there be one `arguments`?

Reviewer 2: Hi @MichelleArk , can you please confirm if this is correct?

Author: This is correct. It's really a niche edge case that users should only hit if they happen to have a custom argument already named 'arguments'. This would be the way to preserve backward-compatibility in that case. We could also mention to consider renaming `arguments` to something else + nest it under `arguments` as a workaround as that would be less confusing

Author: updated

The following are deprecation warnings in dbt today and the associated version number in which they first appear.

### ArgumentsPropertyInGenericTestDeprecation

dbt has deprecated the ability to specify a custom top-level property called `arguments` on generic tests. This deprecation warning is only raised when the behavior flag `require_generic_test_arguments_property` is set to `False`.

#### ArgumentsPropertyInGenericTestDeprecation warning resolution

> If you previously had a property called `arguments` on custom generic tests:

Reviewer: ```suggestion
For example, you may have previously had a property called `arguments` on custom generic tests:
```

<File name='model.yml'>

```yaml
models:
  - name: my_model_with_generic_test
    data_tests:
      - my_custom_generic_test:
          arguments: [1,2,3]
          expression: "order_items_subtotal = subtotal"
```

</File>

> You should flip the `require_generic_test_arguments_property` flag to `True` and nest any keyword arguments to your test under the new `arguments` property:

Reviewer: ```suggestion
You should set the `require_generic_test_arguments_property` flag to `True` and nest any keyword arguments to your test under the new `arguments` property:
```

The following are deprecation warnings in dbt today and the associated version number in which they first appear.

### ArgumentsPropertyInGenericTestDeprecation

dbt has deprecated the ability to specify a custom top-level property called `arguments` on generic tests. This deprecation warning is only raised when the behavior flag `require_generic_test_arguments_property` is set to `False`.

> 

Reviewer: small commit to fix Vercel failure
```suggestion
```

This deprecation warning is only raised when the behavior flag `require_generic_test_arguments_property` is set to `True`.

#### MissingArgumentsPropertyInGenericTestDeprecation warning resolution

If you previously set arguments as top-level properties on custom generic tests:

<File name='model.yml'>

```yaml
models:
  - name: my_model_with_generic_test
    data_tests:
      - dbt_utils.expression_is_true:
          expression: "order_items_subtotal = subtotal"
```

</File>

Or using the alternative `test_name` format:

<File name='model.yml'>

```yaml
models:
  - name: my_model_with_generic_test
    data_tests:
    - name: arbitrary_name
      test_name: dbt_utils.expression_is_true
      expression: "order_items_subtotal = subtotal"
      where: "1=1"
```

</File>

You should now nest arguments under `arguments` and framework configurations under `config`:

<File name='model.yml'>

```yaml
models:
  - name: my_model_with_generic_test
    data_tests:
      - dbt_utils.expression_is_true:
          arguments: 
            expression: "order_items_subtotal = subtotal"
```

</File>

> Or:

Reviewer: Should this say something like this?
```suggestion
Or with framework configurations:
```
